src/hotspot/share/opto/subnode.cpp

```diff
@@ -1467,27 +1484,69 @@ static bool is_counted_loop_cmp(Node *cmp) {
          n->in(0)->as_CountedLoop()->phi() == n;
 }
 
+static bool is_arithmetic_cmp(Node* cmp) {
+  if (!cmp->is_Cmp()) {
```

Comment:
Hi Cesar, I receive feedback from Roland in #13039 that suggested extending identical_backtoback_ifs instead of adding Ideal for BoolNode and CmpNode. 

Before making any modifications based on your review comment, I would first like to reach a consensus on whether we should add idealization for bool and cmp or extend identical_backtoback_ifs. Do you have any comments in this regard? Perhaps @TobiHartmann could also chime in.

Reply:
I was unaware of the other PR. Please ignore my comments if the other PR covers this work.